### PR TITLE
Add logging for some method entry points

### PR DESCRIPTION
**Pull request by warniaha**
Add more method entry point logging

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sal_daos.cc [hunk 1]
```diff
@@ -1950,9 +1981,13 @@ DaosAtomicWriter::DaosAtomicWriter(
       ptail_placement_rule(_ptail_placement_rule),
       olh_epoch(_olh_epoch),
       unique_tag(_unique_tag),
-      obj(_store, _head_obj->get_key(), _head_obj->get_bucket()) {}
+      obj(_store, _head_obj->get_key(), _head_obj->get_bucket())
+{
+  ldpp_dout(dpp, 20) << "DEBUG: DaosAtomicWriter::DaosAtomicWriter" << dendl;
+}
```
Comment (zalsader) [inline src/rgw/rgw_sal_daos.cc:1987]: This is just a constructor, is it necessary?

File: src/rgw/rgw_sal_daos.h [hunk 2]
```diff
@@ -697,7 +700,9 @@ class DaosObject : public Object {
 class MPDaosSerializer : public MPSerializer {
  public:
   MPDaosSerializer(const DoutPrefixProvider* dpp, DaosStore* store,
-                   DaosObject* obj, const std::string& lock_name) {}
+                   DaosObject* obj, const std::string& lock_name) {
+    DAOS_NOT_IMPLEMENTED_LOG(dpp);
+  }
```
Comment (zalsader) [inline src/rgw/rgw_sal_daos.h:705]: This is just a constructor, is it necessary?

**Review by zalsader — APPROVED** [review]
Looks good

File: src/rgw/rgw_sal_daos.cc (hunk 1 above)
Comment (warniaha) [reply src/rgw/rgw_sal_daos.cc:1987]: you're right.  I guess I got too happy with adding those

**Pull request merged**